src/test/java/bio/terra/service/dataset/flight/update/DatasetSchemaUpdateValidationTest.java

```diff
@@ -111,5 +119,92 @@ public void testValidations() throws Exception {
+                                            .datatype(TableDataType.STRING),
+                                        new ColumnModel()
+                                            .name("new_column")
+                                            .datatype(TableDataType.STRING))),
+                            new DatasetSchemaColumnUpdateModel()
+                                .tableName("new_table")
+                                .columns(
+                                    List.of(
+                                        new ColumnModel()
+                                            .name("new_table_column")
+                                            .datatype(TableDataType.STRING))))));
+
+    DatasetSchemaUpdateValidateModelStep validateModelStep =
+        new DatasetSchemaUpdateValidateModelStep(
+            datasetService, datasetId, duplicateColumnsUpdateModel);
+
+    FlightContext flightContext = mock(FlightContext.class);
+
+    DatasetSchemaUpdateException duplicateColumnsException =
+        (DatasetSchemaUpdateException)
+            validateModelStep.doStep(flightContext).getException().orElseThrow();
+
+    assertThat(
+        duplicateColumnsException.getMessage(),
+        containsString("overwrite existing or to-be-added"));
+    assertThat(duplicateColumnsException.getCauses(), hasSize(2));
+    assertThat(
+        duplicateColumnsException.getCauses(),
+        equalTo(List.of("existing_table:existing_column", "new_table:new_table_column")));
+  }
+
+  @Test
+  public void testColumnMissingTableValidations() throws Exception {
+    DatasetSchemaUpdateModel missingTableUpdateModel =
+        new DatasetSchemaUpdateModel()
+            .changes(
+                new DatasetSchemaUpdateModelChanges()
+                    .addColumns(
+                        List.of(
+                            new DatasetSchemaColumnUpdateModel()
+                                .tableName("not_a_real_table")
+                                .columns(
+                                    List.of(
+                                        new ColumnModel()
+                                            .name("new_column")
+                                            .datatype(TableDataType.STRING))))));
+
+    DatasetSchemaUpdateValidateModelStep validateModelStep =
+        new DatasetSchemaUpdateValidateModelStep(
+            datasetService, datasetId, missingTableUpdateModel);
+
+    FlightContext flightContext = mock(FlightContext.class);
+
+    DatasetSchemaUpdateException missingTableException =
+        (DatasetSchemaUpdateException)
+            validateModelStep.doStep(flightContext).getException().orElseThrow();
+
+    assertThat(missingTableException.getMessage(), containsString("Could not find tables"));
+    assertThat(missingTableException.getCauses(), hasSize(1));
+    assertThat(missingTableException.getCauses().get(0), containsString("not_a_real_table"));
```

Comment:
I think you can do both together, maybe
```suggestion
    assertThat(missingTableException.getCauses(), hasItem(containsString("not_a_real_table")));
```

Reply:
Nice! That works 🙂